src/Indicator.vala

```diff
@@ -164,31 +175,27 @@ public class Sound.Indicator : Wingpanel.Indicator {
         var vol = new Services.VolumeControl.Volume ();
         vol.reason = Services.VolumeControl.VolumeReasons.USER_KEYPRESS;
 
-        int dir = 0;
-        if (e.direction == Gdk.ScrollDirection.UP) {
-            dir = 1;
-        } else if (e.direction == Gdk.ScrollDirection.DOWN) {
-            dir = -1;
-        }
-
-        double v = this.volume_control.volume.volume + volume_step_percentage * dir;
-        vol.volume = v.clamp (0.0, this.max_volume);
-        this.volume_control.volume = vol;
+        double dir = 0.0;
 
-        if (open == false && this.notification != null && v >= -0.05 && v <= (this.max_volume + 0.05)) {
+        if (handle_scroll_event (e, out dir)) {
+            double v = this.volume_control.volume.volume + volume_step_percentage * dir;
+            vol.volume = v.clamp (0.0, this.max_volume);
+            this.volume_control.volume = vol;
 
```

Comment:
Shouldn't we check whether the volume has changed? This would for example prevent notifications and sounds when the volume is already at the max volume.
This is probably what this `v >= -0.05 && v <= (this.max_volume + 0.05)` check did in a weird way.

Reply:
OK, makes sense.